Automattic/wp-calypso
# Add links and tracking to eCommerce trial post-purchase screen

Related to:
* #73500

## Proposed Changes

This PR tackles a few items relating to the post-purchase screen we want to show after users who were on the free trial purchase a paid eCommerce plan:
* We're adding a `PageViewTracker` to the page - this will log `calypso_page_view` events with a `path` value of `/plans/my-plan/trial-upgraded/:site`
* We're adding click tracking and links to the existing cards, where each click is tracked using the `calypso_wooexpress_trial_upgraded_card_click` Tracks event, with the card identified by the task's `id` value - the specifics are:
   - _Select your custom domain_ - `custom-domain` - Add a domain flow - `/domains/add/:site?suggestion=:siteName`
   - _Promote your products_ - `promote-products` - Marketing task in onboarding - `/wp-admin/admin.php?page=wc-admin&task=marketing`
   - _Provide a way to pay_ - `payment-methods` - Payments task in onboarding - `/wp-admin/admin.php?page=wc-admin&task=payments`
   - _Create a marketing campaign_ - `marketing-campaign` - AutomateWoo presets screen - `/wp-admin/edit.php?post_type=aw_workflow#presets`
      * **NOTE:** This is a departure from the design, as we are not installing MailPoet - FYI @verofasulo, @manospsyx 
   - _Make your store stand out_ - `customize-store` - WordPress Site Editor - `/wp-admin/site-editor.php`
      * Not sure if it's 100% safe to open the site editor if we don't have an FSE-compatible theme
   - _Manage your store on the go_ - `woocommerce-app` - mobile app modal in onboarding - `/wp-admin/admin.php?page=wc-admin&mobileAppModal=true`
      * I had some trouble testing the modal on my sites; I am also not sure there is a better target for this card

## Testing Instructions

* Ensure that you have Tracks Vigilante installed for inspecting Tracks events - see https://github.com/Automattic/tracks-chrome-extension for more details
* Pick an eCommerce site you already have (or upgrade a site)
* Run this branch locally or via Calypso.live
* Navigate to `/plans/my-plan/trial-upgraded/:siteSlug`
* Verify that you see a `calypso_page_view` event with `path` = `/plans/my-plan/trial-upgraded/:site`
* Mouse over all of the cards, and confirm that their `href` values point at the expected targets in the change details above
* Click on each of the cards, and verify that it takes you to complete the task it described
* Also verify that each click results in a `calypso_wooexpress_trial_upgraded_card_click` Tracks event with a `card_id` prop set to the value of the card's identifier/slug listed above.
* Verify that the padding and behaviour also work as expected on mobile screen sizes

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
